File: marvelHeroes/src/main/java/com/futureworkshops/marvelheroes/presentation/splashscreen/SplashscreenPresenter.java
/*
 * Copyright (c) 2018 FutureWorkshops. All rights reserved.
 */

package com.futureworkshops.marvelheroes.presentation.splashscreen;

import com.futureworkshops.marvelheroes.domain.navigator.Navigator;

import javax.inject.Inject;


public class SplashscreenPresenter implements SplashscreenContract.Presenter {
    
    private Navigator navigator;
    private final SplashscreenContract.View viewReference;
    
    @Inject
    public SplashscreenPresenter(Navigator navigator, SplashscreenContract.View view) {
        this.navigator = navigator;
        this.viewReference = view;
    }
    
    @Override
    public void showCharacterScreen() {
        navigator.showCharacterScreen();
    }
}

File: marvelHeroes/src/main/java/com/futureworkshops/marvelheroes/presentation/splashscreen/SplashScreenPresenter.kt
/*
 * Copyright (c) 2018 FutureWorkshops. All rights reserved.
 */

package com.futureworkshops.marvelheroes.presentation.splashscreen

import com.futureworkshops.marvelheroes.domain.navigator.Navigator

import javax.inject.Inject


class SplashScreenPresenter
@Inject
constructor(private val navigator: Navigator, private val viewReference: SplashScreenContract.View) : SplashScreenContract.Presenter {
    
    override fun showCharacterScreen() {
        navigator.showCharactersScreen()
    }
}
